### ISSUE TITLE
Availability of Views ui.hash in Behaviors is inconsistent

### ISSUE THREAD

**Issue by benmccormick** [marionettejs/backbone.marionette#2339]
From my discussion with @samccone on Gitter:

The way Behaviors mixin Views ui.hashes is inconsistent right now.

On the events and trigger hashes, Behaviors mixin `view.ui` to their own independent ui hash.  In callback functions, only the Behavior's ui hash is available.

So this works.

```javascript
let ExampleBehavior = Mn.Behavior.extend({
  events: {
    'click @ui.test' : 'doStuff'
  },

  doStuff: function() {
   //...
  }
});

let ExampleView = Mn.ItemView.extend({
  ui: {
   test: '.test'
  },
  behaviors: {
    example: {
      behaviorClass: ExampleBehavior
    }
  }

});
```

and this works

```javascript
let ExampleBehavior = Mn.Behavior.extend({
  events: {
    'click @ui.test' : 'doStuff'
  },
  
  ui: {
    test: '.test'
  },

  doStuff: function() {
    this.ui.test.addClass('example');
  }
});

let ExampleView = Mn.ItemView.extend({
 
  behaviors: {
    example: {
      behaviorClass: ExampleBehavior
    }
  }

});
```

but this does not

```javascript
let ExampleBehavior = Mn.Behavior.extend({
  events: {
    'click @ui.test' : 'doStuff'
  },
  
  doStuff: function() {
    this.ui.test.addClass('example');
  }
});

let ExampleView = Mn.ItemView.extend({
   ui: {
    test: '.test'
  },
  behaviors: {
    example: {
      behaviorClass: ExampleBehavior
    }
  }

});
```

It would be great to make this behavior more consistent, and either mix in `view.ui` consistently, or ignore it entirely.

Related to #2338

**Comment by samccone** [comment]
:+1: 

the only thing... I dislike about the coupling is that ideally behaviors should be "dumb" and not know about your views too much... this encourages coupling but.. I suppose we are already quite far down that hole

**Comment by benmccormick** [comment]
I see this as making them **less knowledgeable** about views because they don't have to know the selectors/structure of the view (which is needed if you're going to use selectors in a separate behavior hash).  It does mean views need to know more about the behaviors, but I'd rather have the views knowing about the behaviors than vice-versa.

**Comment by samccone** [comment]
yep i'm with you :)

**Comment by ahumphreys87** [comment]
Fixed in 2.4 via #2344
